kivy/core/window/_window_sdl2.pyx

```diff
@@ -686,3 +707,36 @@ cdef SDL_Surface* flipVert(SDL_Surface* sfc):
         memcpy(&rpixels[pos], &pixels[(pxlength - pos) - pitch], pitch)
 
     return result
+
+cdef SDL_Surface* surface_from_core_image(core_image):
+    # Create an SDL_Surface from a core image's texture pixels. Use
+    # SDL_FreeSurface(surface) to deallocate.
+    #
+    # Returns NULL if the core image has no texture pixels, or if
+    # there's a problem with the creation of an SDL Surface.
+    cdef int width, height
+    width, height = core_image.size
+    cdef int pitch = width * 4
+    cdef bytes pixels
+    cdef char* sdl_error
+    try:
+        pixels = core_image.image.texture.pixels
+    except AttributeError:
+        return NULL
+
+    if pixels == None:
+        return NULL
+
+    cdef Uint8 *c_pixels = pixels
+    cdef SDL_Surface *surface
+    surface = SDL_CreateRGBSurface(0, width, height, 32,
```

Comment:
Seems like you could use `SDL_CreateRGBSurfaceFrom` here instead of the additional memcpy

Reply:
Alright, I've looked at it again.

I started out with an `SDL_CreateRGBSurfaceFrom()` call but was unable to make that work. My previous attempt looked like this:
```cython
    cdef char *c_pixels = core_image.image.texture.pixels
    cdef SDL_Surface *icon_surface
    icon_surface = SDL_CreateRGBSurfaceFrom(c_pixels, width, height, 32,
        pitch, 0x00000000ff, 0x0000ff00, 0x00ff0000, 0xff000000)
    return icon_surface
```

`SDL_CreateRGBSurface()` allocates the necessary buffer and `memcpy()` below copies the pixels into it, so that should work correctly. Maybe I'm not getting something, but how would `SDL_CreateRGBSurfaceFrom()` help here? I tried that, where's the error?